doc/user_mgmt/user_mgmt_hld.md

```diff
@@ -0,0 +1,595 @@
+It will define the default roles (capabilities), default users and describe the supported configurations
+for users.<br/>
+
+Note:
+The compilition of the feature code will be controlled by a build flag and it will be disabled by default.
+
+###  1.5. <a name='Requirements'></a>Requirements
+####  1.5.1. <a name='Definedefaults'></a>Define defaults
+* Default roles: admin and monitor<br/>
+* Default users: admin and monitor
+####  1.5.2. <a name='Configurations'></a>Configurations
+* **add/delete user**
+
+	Add/Delete a user from system. Default users shouldn't be deleted.<br/>
+    Defining the password is mandatory when adding a new user.
+* **enable/disable user**
+
+	Control login of the user by enabling/disabling user state.
+* **change full-name**
+
+	Change Gecos field (Description) of user
+* **change role (capability)**
+
+	Control the capability of user by changing the role
+* **change password**
+
+	Change the password of a user.
+
+##  2 <a name='Design'></a>Design
+
+###  2.1 <a name='ArchitectureDesign'></a>Architecture Design
+![Arch_diagram](Arch_diagram.png)
+
+(flow description in the chapter below)
+
+###  2.2 <a name='High-LevelDesign'></a>High-Level Design
+
+Sonic has two main users:
+* Admin: Administrator with limited access rights.It is part of sudo group.<br/>
+* Root: Superuser with highest access rights. Login is disabled for this user.
+
+We will define two roles for users:<br/>
+* **admin**: a role for administrator users with write and read permissions.<br/>
+	Each admin user will be part of the following groups:<br/>
+	**primary group**: admin<br/>
+	**secondary groups**: sudo,docker,redis,adm
+
+* **monitor**: a role for monitoring users with read permissions only. <br/>
+	Each monitor user will be part of the following groups:<br/>
+	**primary group**: adm<br/>
+	**secondary groups**: N/A
+
+In addition, we will define two default users:
+* **admin**: User role is admin.<br/>
+	The only change from the current admin in Sonic is the additional group: adm
+
+* **monitor**: User role is monitor. <br/>
+	User with monitoring permissions only .<br/>
+
+**Note**:
+ Group adm is used for system monitoring tasks. Members of this group can read the files under "/var/log"
+
+
+###  2.3 <a name='UsercfdRole'></a>Usercfgd Role:
+
+Usercfgd is the daemon that get events from USER_TABLE in CONF_DB and apply the new configuration on the system.<br/>
+The new daemon should start running after "config-setup.service" (i.e. after database is ready) but before systemd enables "system-logind.service" (i.e. before user can login to system)<br/>
+
+
+![services_diagram](services_diagram.png))<br/>
+
+
+##  3 <a name='Configurationandmanagement'></a>Configuration and management
+
+###  3.1 <a name='ConfigDBTables'></a>ConfigDB Tables
+
+```
+USER_TABLE:{
+  "<username>":{
+    "state": {{enabled/disabled}}
+    "full-name": {{string}}
+    "hashed-password": {{string}}
```

Comment:
+1 agree with @jeffhtgt 's comment above. The hashed user pw is already stored in `/etc/shadow`, which is permission-restricted in the Linux file system. Putting the hashed pw in CONFIG_DB will circumvent those permission restrictions.

Reply:
I agree. 
I am working with the team on a new design for storing users passwords.